library/ssl_tls.c

```diff
@@ -6200,6 +6189,7 @@ static int ssl_session_save( const mbedtls_ssl_session *session,
     }
 #endif /* MBEDTLS_SSL_PROTO_TLS1_2_OR_EARLIER */
 
+#if defined(MBEDTLS_SSL_PROTO_TLS1_2_OR_EARLIER)
```

Comment:
This is not enough, we also need to check the TLS version in _use_ here. But let's fix it another time.